aws/ec2/snappy_ec2.py

```diff
@@ -1527,11 +1539,30 @@ def real_main():
               "on the local file system", file=stderr)
         sys.exit(1)
 
+    if not (opts.assume_role_arn is None):    
+        sts_client = boto3.client('sts')
+        response = sts_client.assume_role(
+            DurationSeconds=opts.assume_role_timeout,
+            RoleArn=opts.assume_role_arn,
+            RoleSessionName=opts.assume_role_session_name)
+    
     try:
         if opts.profile is None:
-            conn = ec2.connect_to_region(opts.region)
+            if opts.assume_role_arn is None:
+                conn = ec2.connect_to_region(opts.region)
+            else:
+                conn = boto.ec2.connect_to_region(opts.region, aws_access_key_id=response['Credentials']['AccessKeyId'], 
+                        aws_secret_access_key=response['Credentials']['SecretAccessKey'],
+                        security_token=response['Credentials']['SessionToken'])
         else:
-            conn = ec2.connect_to_region(opts.region, profile_name=opts.profile)
+            if opts.assume_role_arn is None:
+                conn = ec2.connect_to_region(opts.region, profile_name=opts.profile)
+            else:
+                conn = boto.ec2.connect_to_region(opts.region, profile_name=opts.profile, 
```

Comment:
Are there any other places where we need to put the assumed-role's credentials? Or is this is the only place?
Are the commands start, stop, etc also taken care of apart from the launch and destroy?

Reply:
I have tested the start and stop cluster commands it working fine. I think this is the only place need assume_role call.